chain/chain/src/resharding.rs

```diff
@@ -282,37 +296,33 @@ impl Chain {
         let (snapshot_store, flat_storage_manager) = tries
             .get_state_snapshot(&prev_prev_hash)
             .map_err(|err| StorageInconsistentState(err.to_string()))?;
-        let flat_storage_chunk_view =
-            flat_storage_manager.chunk_view(shard_uid, prev_prev_hash).ok_or_else(|| {
-                StorageInconsistentState("Chunk view missing for snapshot flat storage".to_string())
-            })?;
-        let flat_storage_iter =
-            flat_storage_chunk_view.iter_flat_state_entries(None, None).map(|entry| {
-                let (key, value) = entry.unwrap();
-                (key, Some(value))
-            });
-
+        let flat_storage_chunk_view = flat_storage_manager.chunk_view(shard_uid, prev_prev_hash);
+        let flat_storage_chunk_view = flat_storage_chunk_view.ok_or_else(|| {
+            StorageInconsistentState("Chunk view missing for snapshot flat storage".to_string())
+        })?;
+        // Get the flat storage iter and wrap the value in Optional::Some to
+        // match the delta iterator so that they can be chained.
+        let flat_storage_iter = flat_storage_chunk_view.iter_flat_state_entries(None, None);
+        let flat_storage_iter = flat_storage_iter.map_ok(|(key, value)| (key, Some(value)));
+
+        // Get the delta iter and wrap the items in Result to match the flat
+        // storage iter so that they can be chained.
         let delta = store_helper::get_delta_changes(&snapshot_store, shard_uid, prev_hash)
-            .map_err(|err| StorageInconsistentState(err.to_string()))?
-            .ok_or_else(|| {
-                StorageInconsistentState("Delta missing for snapshot flat storage".to_string())
-            })?;
+            .map_err(|err| StorageInconsistentState(err.to_string()))?;
+        let delta = delta.ok_or_else(|| {
+            StorageInconsistentState("Delta missing for snapshot flat storage".to_string())
+        })?;
         let delta_iter = delta.0.into_iter();
+        let delta_iter = delta_iter.map(|item| Ok(item));
```

Comment:
whoa. are value overwrites for readability?

Reply:
Yep, I like it more this way. One line = one statement. It's quite rusty too.